### ISSUE TITLE
Proper value for `responseType` is not set when response type is a `binary` string

### ISSUE THREAD

**Issue by usersina** [anymaniax/orval#870]
**TL;DR:** A `responseType` of `blob` is missing when the return type of an endpoint is a single file (Or a [binary string](https://swagger.io/docs/specification/data-models/data-types/#file)) which messes with the file encoding.

## What are the steps to reproduce this issue?

Given the following (partial) schema, generate the `client` types

```json
{
  // ...
  "/file/preview": {
    "get": {
      "tags": ["file-controller"],
      "operationId": "downloadFile",
      "responses": {
        "default": {
          "description": "default response",
          "content": {
            "*/*": {
              "schema": {
                "type": "string",
                "format": "binary"
              }
            }
          }
        }
      }
    }
  }
}
```

## What happens?

For the `downloadFile` function using `client: "react-query"`, this is what's generated

```typescript
export const useDownloadFileHook = () => {
  const downloadFile = useCustomInstance<unknown>(); // notice the unknown type, should be `Blob`

  return (signal?: AbortSignal) => {
    return downloadFile({ method: "get", signal });
  };
};
```

However, this is broken and the fetched file will be encoded in the wrong way (e.g. an image will now show up).
In addition to that, the type is set to `unknown` instead of `Blob`.

```typescript
const downloadFile = useDownloadFileHook();
const data = await downloadFile();

console.log(URL.createObjectURL(new File([data as Blob]))); // blob:http://localhost:3000/hardcoded-file
```

The reason for this, is that the called `downloadFile` does not specify the `responseType: "blob"`, even though the schema specifies that it's a binary string.

## What were you expecting to happen?

Instead, this is what works if I manually update the file

```typescript
export const useDownloadFileHook = () => {
  const loadSpecificFileAsPreview = useCustomInstance<Blob>(); // for type safety

  return (signal?: AbortSignal) => {
    return downloadFile({ method: "get", signal, responseType: "blob" });
  };
};
```

## Any other comments?

For now, I fetch the file using my own custom instance:

```typescript
const customAxios = useCustomInstance<Blob>();

 const data = await customAxios({
  method: "get",
  url: "/file",
  responseType: "blob", // This line solves the issue
});
```

## What versions are you using?
**Package Version: `^6.15.0`**

**Comment by yuta-ike** [comment]
@usersina 

> In addition to that, the type is set to unknown instead of Blob.

This does not seem to be limited to `blob`. For example, even with the following schema, the response is unknown. Maybe this is due to the use of `default` as the HTTP status of the response.

```yml
/file/preview:
  get:
    tags:
      - file-controller
    operationId: downloadFile
    responses:
      default:
        description: default response
        content:
          application/json:
            schema:
              type: integer
```

How about using "200" instead?
I don't think `default` is supported in Orval...

**Comment by usersina** [comment]
@yuta-ike

Sorry for the late reply, but that actually solves the problem! Thank you!
A bit curious as to what might the be reason of Orval not supporting `default`?

For the record, I had a spring backend and I have to explicitly add the status code to the `@ApiResponse` to make it work.

```java
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;

@GetMapping(value = "/file/preview", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
@ApiResponse(content = @Content(schema = @Schema(type = "string", format = "binary")), responseCode = "200")
public ResponseEntity<byte[]> downloadFile( ...
```

**Comment by yuta-ike** [comment]
@usersina 
I don't know the actual reason because I'm not a maintainer of this project. But the good news is that my pr, which changed to use `default`, has been merged. Please update to the latest version and give it a try again.

https://github.com/anymaniax/orval/pull/877

**Comment by usersina** [comment]
Thanks for the PR, it works now! Closing this issue.
